ssl/record/rec_layer_s3.c

```diff
@@ -1422,6 +1422,20 @@ int ssl3_read_bytes(SSL *s, int type, int *recvd_type, unsigned char *buf,
             if (SSL3_RECORD_get_length(rr) == 0)
                 SSL3_RECORD_set_read(rr);
 
+            if (SSL_IS_TLS13(s)
```

Comment:
NSS and BoringSSL both do this at all versions, by the way. So you'd be in good company if you enforced it unconditionally. That would let you avoid the `alert_fragment` logic on the TLS side. On the DTLS side, it ought to be removed anyway because fragmented alerts wouldn't work at all in DTLS.